test/SourceKit/Indexing/index_locals.swift

```diff
@@ -0,0 +1,86 @@
+// RUN: %empty-directory(%t)
+// RUN: %sourcekitd-test -req=index-to-store %s -index-store-path %t/idx -index-unit-output-path %t/index_locals.o -req-opts=include_locals=1 -- %s
+// RUN: c-index-test core -print-record %t/idx | %FileCheck %s
```

Comment:
IMO it's enough to just test that a single local is indexed here, assuming we have a test for locals themselves in test/Index. If we don't then I'd prefer most of that going there and then a simple test here to make sure the option itself works.